Feat/router and tailwind

parradaniela opened the pull request:
Tailwind: 
- Installed Tailwind, removed default CSS that comes in App.css and index.css
- Introduced custom wrapper class in index.css to keep everything centered and with a max width

Router: 
- Set up Router with a root Layout component and a nested About route
- RootLayout includes a header and footer that will wrap around all nested routes
- Added some basic tailwind flex classes to components so that they reflect the style in the wireframe

All changes are in their own folders and shouldn't cause conflicts with others' work, though the changes to package.json and package-lock.json will likely cause some merge conflicts with other PRs

Happy to make further edits before merging

parradaniela committed:
Installed Tailwind, removed default CSS styles

parradaniela committed:
Set up Router, root layout and about page Route

parradaniela committed:
Added Landing page component to main route

mays4 reviewed (approved):
I viewed your changes and all nicely organized

parradaniela committed:
Merging with main

parradaniela committed:
Fixing merge conflicts

parradaniela committed:
Removed tailwind, added CSS properties back

parradaniela committed:
Moved Login components to own folder under Pages

parradaniela committed:
Created profile page to add Route

parradaniela committed:
Changed Nav, Footer and RootLayout to use MUI components. Updated styling

parradaniela commented:
Hello all, I updated the PR to include: 
- Resolving merge conflicts with Main branch
- Uninstalled tailwind and removed all tailwind classes 
- Updated Nav, Footer, and RootLayout components to use MUI components
- Moved Login and Form components to their own folder (under Pages) 
- Updated css rules 
- Updated Routes to include Login and Profile pages 

Some questions/comments I have: 
- Is there a specific reason why we have both App.css and index.css? Can we condense to only use one file so that we don't have to check two different files for CSS rules? We should also try as much as possible to keep styling in-line using MUI's sx property
- I had to remove the flex properties from the body element because they were shrinking the entire site to its minimum possible width. Unfortunately this means that the Login form is now much wider than before. Sorry, I wasn't able to fix it tonight but we can fix it later
